### PR TITLE
Fixed resolving for local files

### PR DESCRIPTION
**Pull request by ysrbo**
Fix for [BUG] samm-cli's standard resolver constructs invalid paths instead of resolving prefixes #354

### CODE REVIEW

File: core/esmf-aspect-model-resolver/src/main/java/org/eclipse/esmf/aspectmodel/resolver/AspectModelResolver.java [hunk 1]
```diff
@@ -123,7 +127,7 @@ public Try<VersionedModel> resolveAspectModel( final ResolutionStrategy resoluti
     * @return the resolved model on success
     */
    public Try<VersionedModel> resolveAspectModel( final ResolutionStrategy resolutionStrategy, final InputStream inputStream ) {
-      return TurtleLoader.loadTurtle( inputStream ).flatMap( model -> resolveAspectModel( resolutionStrategy, model ) );
+      return TurtleLoader.loadTurtle( inputStream ).flatMap( model -> resolveAspectModel( FileSystemStrategy.DefaultNamespace.withDefaultNamespace(resolutionStrategy, model), model ) );
```
Comment (atextor) [inline core/esmf-aspect-model-resolver/src/main/java/org/eclipse/esmf/aspectmodel/resolver/AspectModelResolver.java:130]: Adjust code style: There should be spaces inside the brackets. Please also check other locations in this PR.

File: core/esmf-aspect-model-resolver/src/main/java/org/eclipse/esmf/aspectmodel/resolver/FileSystemStrategy.java [hunk 2]
```diff
@@ -63,14 +66,18 @@ public FileSystemStrategy( final Path modelsRoot ) {
...
+    * {@link FileNotFoundException} if no file containing the element was found
     */
    @Override
    public Try<Model> apply( final AspectModelUrn aspectModelUrn ) {
-      final Path directory = modelsRoot.resolve( aspectModelUrn.getNamespace() ).resolve( aspectModelUrn.getVersion() );
+      final Path directory = resolve( aspectModelUrn );
+      return loadFromDirectory( aspectModelUrn, directory );
+   }
+
+   protected Try<Model> loadFromDirectory( AspectModelUrn aspectModelUrn, Path directory ) {
```
Comment (atextor) [inline core/esmf-aspect-model-resolver/src/main/java/org/eclipse/esmf/aspectmodel/resolver/FileSystemStrategy.java:80]: Arguments should be `final`

File: core/esmf-aspect-model-resolver/src/main/java/org/eclipse/esmf/aspectmodel/resolver/FileSystemStrategy.java [hunk 3]
```diff
@@ -90,6 +97,60 @@ public Try<Model> apply( final AspectModelUrn aspectModelUrn ) {
                   .getOrElse( false ) )
             .findFirst()
             .orElse( Try.failure( new FileNotFoundException(
-                  "No model file containing " + aspectModelUrn.toString() + " could be found in directory: " + directory ) ) );
+                  "No model file containing " + aspectModelUrn + " could be found in directory: " + directory ) ) );
+   }
+
+   protected Path resolve( AspectModelUrn aspectModelUrn ) {
```
Comment (atextor) [inline core/esmf-aspect-model-resolver/src/main/java/org/eclipse/esmf/aspectmodel/resolver/FileSystemStrategy.java:103]: Arguments should be `final`

File: core/esmf-aspect-model-resolver/src/main/java/org/eclipse/esmf/aspectmodel/resolver/FileSystemStrategy.java [hunk 4]
```diff
@@ -90,6 +97,60 @@ public Try<Model> apply( final AspectModelUrn aspectModelUrn ) {
...
+      return "FileSystemStrategy(root=" + modelsRoot + ')';
+   }
+
+   /**
+    * Initialize the File System Strategy where the default namespace is resolved in the root folder.
+    */
+   public static class DefaultNamespace extends FileSystemStrategy {
+      AspectModelUrn defaultUrn;
+
+      public DefaultNamespace( Path modelsRoot ) {
```
Comment (atextor) [inline core/esmf-aspect-model-resolver/src/main/java/org/eclipse/esmf/aspectmodel/resolver/FileSystemStrategy.java:118]: Arguments should be `final`

File: core/esmf-aspect-model-resolver/src/main/java/org/eclipse/esmf/aspectmodel/resolver/FileSystemStrategy.java [hunk 5]
```diff
@@ -90,6 +97,60 @@ public Try<Model> apply( final AspectModelUrn aspectModelUrn ) {
...
+         super( modelsRoot );
+      }
+
+      @Override
+      protected Path resolve( AspectModelUrn urn ) {
```
Comment (atextor) [inline core/esmf-aspect-model-resolver/src/main/java/org/eclipse/esmf/aspectmodel/resolver/FileSystemStrategy.java:123]: Arguments should be `final`

File: core/esmf-aspect-model-resolver/src/main/java/org/eclipse/esmf/aspectmodel/resolver/FileSystemStrategy.java [hunk 6]
```diff
@@ -90,6 +97,60 @@ public Try<Model> apply( final AspectModelUrn aspectModelUrn ) {
...
+         return (urn.getNamespace().equals( defaultUrn.getNamespace() ) && urn.getVersion().equals( defaultUrn.getVersion() ))
+               ? modelsRoot
+               : super.resolve( urn );
+      }
+
+      @Override
+      public Try<Model> apply( final AspectModelUrn urn ) {
```
Comment (atextor) [inline core/esmf-aspect-model-resolver/src/main/java/org/eclipse/esmf/aspectmodel/resolver/FileSystemStrategy.java:130]: Arguments should be `final`

File: core/esmf-aspect-model-resolver/src/main/java/org/eclipse/esmf/aspectmodel/resolver/FileSystemStrategy.java [hunk 7]
```diff
@@ -90,6 +97,60 @@ public Try<Model> apply( final AspectModelUrn aspectModelUrn ) {
...
+         return loadFromDirectory( urn, directory )
+               .recoverWith( FileNotFoundException.class,
+                     ex -> loadFromDirectory( urn, super.resolve( urn ) )
+                           .mapFailure( Case( $( instanceOf( FileNotFoundException.class ) ),
+                                 e -> new FileNotFoundException( ex.getMessage() + ". AND " + e.getMessage() ) ) )
+               );
+
+      }
+
+      public static <T extends ResolutionStrategy> T withDefaultNamespace( T strategy, Model model ) {
```
Comment (atextor) [inline core/esmf-aspect-model-resolver/src/main/java/org/eclipse/esmf/aspectmodel/resolver/FileSystemStrategy.java:141]: Arguments should be `final`

**Review by atextor — CHANGES_REQUESTED** [review]

File: core/esmf-aspect-model-resolver/src/main/java/org/eclipse/esmf/aspectmodel/resolver/AspectModelResolver.java [hunk 8]
```diff
@@ -13,6 +13,9 @@
 
 package org.eclipse.esmf.aspectmodel.resolver;
 
+import static io.vavr.API.*;
```
Comment (Yauhenikapl) [inline core/esmf-aspect-model-resolver/src/main/java/org/eclipse/esmf/aspectmodel/resolver/AspectModelResolver.java:16]: I propose to use single import without ".*"

But if you use all functionality from this package, don't need.

See for yourself

**Review by Yauhenikapl — APPROVED** [review]

**Review by atextor — APPROVED** [review]

**Pull request merged**